From c05dd814731c2fa41320a575fc4136a131a5f482 Mon Sep 17 00:00:00 2001
From: Gao Pan <b54642@freescale.com>
Date: Tue, 13 Oct 2015 18:27:45 +0800
Subject: [PATCH 0573/5242] MLK-11602 arm: imx: set imx6qdl eim_slow clk to
 135Mhz

commit  5dda5ff17ad17b6eecee8fca334c51c4228c85f3 from
https://source.codeaurora.org/external/imx/linux-imx.git

weim nor read performance drop 32% compared with L3.10.53_1.1.0_GA because
eim_slow clk rate drops. This patch set the eim_slow clk to 135Mhz to improve
the performance.

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Gao Pan <b54642@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx6q.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/imx/clk-imx6q.c b/drivers/clk/imx/clk-imx6q.c
index 0f1414c..6bd1a51 100644
--- a/drivers/clk/imx/clk-imx6q.c
+++ b/drivers/clk/imx/clk-imx6q.c
@@ -901,6 +901,10 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 	clk_set_parent(clk[IMX6QDL_CLK_IPU2_DI0_SEL], clk[IMX6QDL_CLK_IPU2_DI0_PRE]);
 	clk_set_parent(clk[IMX6QDL_CLK_IPU2_DI1_SEL], clk[IMX6QDL_CLK_IPU2_DI1_PRE]);
 
+	imx_clk_set_parent(clk[IMX6QDL_CLK_AXI_ALT_SEL], clk[IMX6QDL_CLK_PLL3_PFD1_540M]);
+	imx_clk_set_parent(clk[IMX6QDL_CLK_AXI_SEL], clk[IMX6QDL_CLK_AXI_ALT_SEL]);
+
+
 	/*
 	 * The gpmi needs 100MHz frequency in the EDO/Sync mode,
 	 * We can not get the 100MHz from the pll2_pfd0_352m.
@@ -957,8 +961,8 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 	if (IS_ENABLED(CONFIG_PCI_IMX6))
 		clk_set_parent(clk[IMX6QDL_CLK_LVDS1_SEL], clk[IMX6QDL_CLK_SATA_REF_100M]);
 
-	/* set eim_slow to 132Mhz */
-	imx_clk_set_rate(clk[IMX6QDL_CLK_EIM_SLOW], 132000000);
+	/* set eim_slow to 135Mhz */
+	imx_clk_set_rate(clk[IMX6QDL_CLK_EIM_SLOW], 135000000);
 
 	/*
 	 * Enable clocks only after both parent and rate are all initialized
-- 
1.7.9.5

